fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -673,30 +650,26 @@ impl<'vm, S, I> CodeCopyCtx<'vm, S, I> {
         I: Iterator<Item = &'vm ContractId>,
         S: InterpreterStorage,
     {
-        let contract = CheckedMemConstLen::<{ ContractId::LEN }>::new(b)?;
-        let cd = checked_add_word(c, d)?;
-
-        if d > MEM_MAX_ACCESS_SIZE
-            || a > checked_sub_word(VM_MAX_RAM, d)?
-            || cd > VM_MAX_RAM
-        {
+        let contract_id = ContractId::from(read_bytes(self.memory, b)?);
+        let offset: usize = c.try_into().map_err(|_| PanicReason::MemoryOverflow)?;
+        // Check target memory range ownership
+        if !self.owner.has_ownership_range(&MemoryRange::new(a, d)?) {
             return Err(PanicReason::MemoryOverflow.into())
         }
 
-        let (a, c, d) = (a as usize, c as usize, d as usize);
-        let cd = cd as usize;
-
-        let contract = ContractId::from_bytes_ref(contract.read(self.memory));
+        self.input_contracts.check(&contract_id)?;
 
-        self.input_contracts.check(contract)?;
+        let contract =
+            super::contract::contract(self.storage, &contract_id)?.into_owned();
```

Comment:
It seems we can remove `.into_owned()` because we don't do that for LDC

Reply:
Done in df03c07